### ISSUE TITLE
Using of AzureAppConfigurationRefreshMiddleware breaks durable function orhestration

### ISSUE THREAD

**Issue by ivankaurov** [Azure/AppConfiguration-DotnetProvider#484]
Code of [AzureAppConfigurationRefreshMiddleware](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/e6988fad3e51c59af6b3800d917f43aa7a265c29/src/Microsoft.Azure.AppConfiguration.Functions.Worker/AzureAppConfigurationRefreshMiddleware.cs#L51C32-L51C55) includes the following line
```cs
await next(context).ConfigureAwait(false);
```
This line breaks the execution of Durable Orchestrator by leaving it in `running` state even after the execution is completed succesfully. It  means that `Microsoft.Azure.AppConfiguration.Functions.Worker` package and especially `AzureAppConfigurationRefreshExtensions.UseAzureAppConfiguration` can't be used with Durable out-of-process Azure functions.

**Comment by jimmyca15** [comment]
@ivankaurov

It sounds then that our library needs to know if it's running on a platform that doesn't support `ConfigureAwait`. Up until this point the adopted practice has been always use `ConfigureAwait`. My initial thinking is that we'll need to figure out a way to detect that our library is running in this environment and have some helper to decide whether to configure await or not.

Thank you for reporting !

@amerjusupovic

**Comment by jviau** [comment]
I don't believe there is anything for AppConfiguration team to do here. The issue lies entirely on the durable side, we are imposing a constraint on the rest of the functions middleware pipeline - which frankly is not a good design on our end. There is an issue I opened to provide an extensibility point to for durable to avoid the issue: https://github.com/Azure/azure-functions-dotnet-worker/issues/1666

**Comment by amerjusupovic** [comment]
https://github.com/Azure/AppConfiguration/issues/915

**Comment by JonathanNagy** [comment]
I came up with a "synchronous" version of the AppConfigurationRefreshMiddleware. It works, but I am reluctant to use it as I don't want to worry about who will maintain it if the AzureAppConfiguration library is updated. I'd be nice if this were fixed, but I can live with having to restart the app function to refresh the configuration.

**Comment by jviau** [comment]
@samsadsam you can close this issue. It is an issue with durable, not app config.

**Comment by jimmyca15** [comment]
@jviau I agree and it looks like there is an issue that tracks that [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/1666). I will close this.

**Comment by kemmis** [comment]
> I don't believe there is anything for AppConfiguration team to do here. The issue lies entirely on the durable side, we are imposing a constraint on the rest of the functions middleware pipeline - which frankly is not a good design on our end. There is an issue I opened to provide an extensibility point to for durable to avoid the issue: [Azure/azure-functions-dotnet-worker#1666](https://github.com/Azure/azure-functions-dotnet-worker/issues/1666)

Looks like this was implemented in the fn worker library last year. I'm still having issues getting app config values to refresh automatically in my fn app... at least using Microsoft.Azure.AppConfiguration.Functions.Worker 8.2.0. I'm still having to use a custom middleware workaround to get app config values to update in my app. Is there still work to be done on the App Config lib side now that the fn worker lib implemented item 1666?
